### ISSUE TITLE
prefix_is_current_class: does not detect me->attribute

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#2033]
```abap
REPORT zfoobar.

CLASS lcl_test DEFINITION.
  PRIVATE SECTION.
    DATA i TYPE i.
    METHODS m.
ENDCLASS.

CLASS lcl_test IMPLEMENTATION.
  METHOD m.
    me->m( ).  "<<< error
    me->i = 1. "<<< no issue
  ENDMETHOD.
ENDCLASS.
```

Finds first `me` but not the second one `me->i`. 
/zfoobar.prog.abap[11, 5]: Omit 'me->' in instance calls(prefix_is_current_class)

**Comment by larshp** [comment]
thanks

**Comment by larshp** [comment]
probably as a new parameter, "omitMeInstanceCalls" is only for calling instance methods

**Comment by mbtools** [comment]
same issue for types:

```abap
CLASS lcl_test DEFINITION.
  PRIVATE SECTION.
    TYPES ty_i TYPE i. "<<< unused
    METHODS m.
ENDCLASS.

CLASS lcl_test IMPLEMENTATION.
  METHOD m.
    DATA lv_1 TYPE me=>ty_i. "<<< no issue
    DATA lv_2 TYPE me->ty_i. "<<< no issue
  ENDMETHOD.
ENDCLASS.
```

also the type is marked as not used when prefixed with `me`:

/zfoobar.prog.abap[4, 11]: Type "ty_i" not used(unused_types)

**Comment by larshp** [comment]
https://rules.abaplint.org/prefix_is_current_class/

**Comment by g-back** [comment]
Note that omitting `me` when accessing attributes can be potentially dangerous if  a method or local variable shadows the attribute name

**Comment by larshp** [comment]
yea, but thats also bad design IMHO

**Comment by g-back** [comment]
That's another matter, I'm just saying that it can happen 🙂
In other languages this is fairly common to use in at least constructors, something like:

```java
public Customer(int customerNumber){
 this.customerNumbeer = customerNumber;
}
```

**Comment by mbtools** [comment]
false positive example where attribute and importing parameter have the same name:

```abap
CLASS lcl_test DEFINITION.
  PUBLIC SECTION.
    METHODS constructor
      IMPORTING
        val TYPE i.
  PRIVATE SECTION.
    DATA val TYPE i.
    METHODS test
      RETURNING
        VALUE(result) TYPE i.
ENDCLASS.

CLASS lcl_test IMPLEMENTATION.
  METHOD constructor.
    me->val = val.     "<<< no error, correct
    me->val = 1.       "<<< no error, correct
    me->val = test( ). "<<< false positive, cannot omit me here
  ENDMETHOD.

  METHOD test.
    result = 1.
  ENDMETHOD.
ENDCLASS.
```
